net-dns/dnsdist/dnsdist-1.1.0.ebuild

```diff
@@ -0,0 +1,134 @@
+# Copyright 1999-2017 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=6
+
+EGIT_REPO_URI="https://github.com/PowerDNS/pdns.git"
+
+if [[ ${PV} = 9999 ]]; then
+	GIT_ECLASS="git-r3"
+	EXPERIMENTAL="true"
+fi
+
+inherit eutils user flag-o-matic systemd autotools ${GIT_ECLASS}
+
+DESCRIPTION="dnsdist is a highly DNS-, DoS- and abuse-aware loadbalancer."
+HOMEPAGE="http://dnsdist.org"
+
+if [[ ${PV} == 9999 ]] ; then
+	SRC_URI=""
+	S="${WORKDIR}/${P}/pdns/dnsdistdist"
+else
+	SRC_URI="https://downloads.powerdns.com/releases/${P}.tar.bz2"
+	KEYWORDS="~amd64"
+fi
+
+LICENSE="GPL-2"
+SLOT="0"
+IUSE="bindist libedit luajit +sodium +protobuf re2 dnscrypt systemd +readline"
+
+RESTRICT="readline? ( bindist )"
+
+REQUIRED_USE="
+	bindist? ( libedit )
+	libedit? ( !readline )
+	readline? ( !libedit )
+	?? ( libedit readline )
+	dnscrypt? ( sodium )
+"
+
+DEPEND="
+	libedit? ( dev-libs/libedit:= )
+	readline? ( sys-libs/readline:0= )
+	>=dev-libs/boost-1.35:=
+	luajit? ( dev-lang/luajit:= )
+	!luajit? ( >=dev-lang/lua-5.1:= )
+	sodium? ( dev-libs/libsodium:= )
+	protobuf? ( dev-libs/protobuf:= )
+	re2? ( dev-libs/re2:= )
+	systemd? ( sys-apps/systemd:= )
+"
+
+RDEPEND="${DEPEND}"
+
+[[ ${PV} == 9999 ]] && DEPEND+="
+	dev-util/ragel
+	sys-devel/automake
+	sys-devel/autoconf
+	app-text/pandoc
+"
+
+src_prepare() {
+	default
+
+	[[ ${PV} == 9999 ]] && eautoreconf
+
+	if use readline ; then
+
+		# Due to licensing (dnsdsit being GPL-2 and readline GPL-3) dnsdist upstream (PowerDNS people)
+		# compiles and links agains libedit (BSD-2), as upstream does provide binary packages itself.
+		# However, dnsdist will work with readline, too, as it does - when compiled from ports on OpenBSD.
+		# Therefore, the default beahavior here is to compile and link against readline, as to not install
+		# the outdated libedit present in the portage tree, requiring a small change to dnsdist.cc and 
+		# dnsdist-console.cc as described in: https://github.com/PowerDNS/pdns/issues/5294
+
+		# sed's --follow-symlinks is a must here for the 9999 version, since those files are placed in ../.
+		# in git, and then symlinked to
+		sed --follow-symlinks -iorig \
```

Comment:
Won't do a backup.

I've first thought of a patch, too, but came across the page in the Gentoo Wiki, where sed in ebuilds is described.

Here is sed simpler, because the includes affected won't change much over time (I hope), and creating patches was more time consuming - also it should be version independent.